jdrfandroidbleparser/src/main/java/org/ehealthinnovation/jdrfandroidbleparser/bgm/characteristic/compoundcharacteristic/GlucoseMeasurementCharacteristic.kt

```diff
@@ -0,0 +1,141 @@
+     * These values are converted into a [Date] as [displayTime]
+     */
+    var timeOffset: Int? = null
+    var timeYear: Int? = null
+    var timeMonth: Int? = null
+    var timeDay: Int? = null
+    var timeHour: Int? = null
+    var timeMinute: Int? = null
+    var timeSecond: Int? = null
+    var displayTime: Date? = null
+
+    /**
+     * The concentration is the raw value from the binary packet. It has unit stored in [unit].
+     * A convenient function [getConcentration] to get the value in desired unit can be used to query this value.
+     */
+    var concentration: Float? = null
+    var unit: Units? = null
+
+    /**
+     * Contains a set of flag concerning the annunciation. Before interpreting the glucose concentration, make sure
+     * no flags indicating measurement error is present in this EnumSet.
+     */
+    var annunciationFlags: EnumSet<SensorStatusAnnunciation>? = null
+
+    /**
+     * The location and type of the blood sample.
+     */
+    var fluidType: Type? = null
+    var sampleLocation: SampleLocation? = null
+
+
+    /**
+     * This function is called when creating an characteristic to parse the binary array stream
+     */
+    override fun parse(c: BluetoothGattCharacteristic): Boolean {
+        var errorFreeParse = false
+        try {
+            //make sure c is not null
+            Flags.parsFlags(getNextIntValue(c, BluetoothGattCharacteristic.FORMAT_UINT8)).let {
+                flags = it
+                sequenceNumber = getNextIntValue(c, BluetoothGattCharacteristic.FORMAT_UINT16)
+                timeYear = getNextIntValue(c, BluetoothGattCharacteristic.FORMAT_UINT16)
+                timeMonth = getNextIntValue(c, BluetoothGattCharacteristic.FORMAT_UINT8)
+                timeDay = getNextIntValue(c, BluetoothGattCharacteristic.FORMAT_UINT8)
+                timeHour = getNextIntValue(c, BluetoothGattCharacteristic.FORMAT_UINT8)
+                timeMinute = getNextIntValue(c, BluetoothGattCharacteristic.FORMAT_UINT8)
+                timeSecond = getNextIntValue(c, BluetoothGattCharacteristic.FORMAT_UINT8)
+
+                if (it.contains(Flags.TIME_OFFSET_PRESENT))
+                    timeOffset = getNextIntValue(c, BluetoothGattCharacteristic.FORMAT_UINT16)
+                if (it.contains(Flags.CONCENTRATION_PRESENT)) {
+                    concentration = getNextFloatValue(c, BluetoothGattCharacteristic.FORMAT_SFLOAT)
+                    val tempIntHolder = getNextIntValue(c, BluetoothGattCharacteristic.FORMAT_UINT8)
+                    fluidType = Type.fromKey(tempIntHolder and 0x0F)
+                    sampleLocation = SampleLocation.fromKey((tempIntHolder and 0xF0) shr 4)
+                    if (it.contains(Flags.CONCENTRATION_UNIT))
+                        unit = Units.MOLE_PER_LITRE
+                    else
+                        unit = Units.KILOGRAM_PER_LITRE
```

Comment:
Either put braces or use a when like here:

`            medicationUnit = when {
                it.contains(Flags.MEDICATION_VALUE_UNITS) -> Units.LITRE
                else -> Units.KILOGRAM
            }`